### PR TITLE
Add `.list.astype()` to cast list leaves to specified dtype

### PR DESCRIPTION
**Pull request by shwina**
Closes https://github.com/rapidsai/cudf/issues/10687.

### CODE REVIEW

File: python/cudf/cudf/tests/test_list.py [hunk 1]
```diff
@@ -759,3 +759,13 @@ def test_listcol_setitem_retain_dtype():
     # prior to this fix: https://github.com/rapidsai/cudf/pull/10151/
     df2 = df1.copy()
     assert df2["a"].dtype == df["a"].dtype
+
+
+def test_list_astype():
```
Comment (rgsl888prabhu) [inline python/cudf/cudf/tests/test_list.py:764]: Should we also compare the leaf values ?

**Review by rgsl888prabhu — APPROVED** [review]
A small query

File: python/cudf/cudf/tests/test_list.py (hunk 1 above)
Comment (shwina) [reply python/cudf/cudf/tests/test_list.py:764]: Fixed - thanks!

**Review by rgsl888prabhu — APPROVED** [review]

**Comment by shwina** [comment]
@gpucibot merge

**Pull request merged**